gnomad_qc/v4/create_release/validate_and_export_vcf.py

```diff
@@ -470,6 +537,317 @@ def prepare_ht_for_validation(
+    :param faf_pops: Dict with faf genentic ancestry group names (keys) and descriptions (values).  Default is FAF_POPS.
+    :param sexes: gnomAD sample sexes used in VCF export. Default is SEXES.
+    :param in_silico_dict: Dictionary of in silico predictor score descriptions.
+    :param vrs_fields_dict: Dictionary with VRS annotations.
+    :param label_delimiter: String to use as delimiter when making group label combinations.
+    :return: Updated INFO dictionary for VCF export.
+    """
+    # Get existing info fields from predefined info_dict, e.g. `FS`,
+    # `non_par`, `negative_train_site`...
+    vcf_info_dict = info_dict.copy()
+    vcf_info_dict = {f: vcf_info_dict[f] for f in info_fields if f in vcf_info_dict}
+
+    # Add allele-specific fields to info dict, including AS_VQSR_FIELDS
+    vcf_info_dict.update(
+        add_as_info_dict(info_dict=info_dict, as_fields=AS_FIELDS + AS_VQSR_FIELDS)
+    )
+
+    for subset in subset_list:
+        subset_pops = deepcopy(pops)
+        if subset == "joint":
+            subset_pops.update({"ami": "Amish"})
+        description_text = "" if subset == "" else f" in {subset} subset"
+
+        vcf_info_dict.update(
+            populate_subset_info_dict(
+                subset=subset,
+                description_text=description_text,
+                pops=subset_pops,
+                faf_pops=faf_pops,
+                sexes=sexes,
+                label_delimiter=label_delimiter,
+            )
+        )
+
+    if age_hist_data:
+        age_hist_data = "|".join(str(x) for x in age_hist_data)
+
+    # Add age histogram data to info dict.
+    vcf_info_dict.update(
+        make_info_dict(
+            label_delimiter=label_delimiter,
+            bin_edges=bin_edges,
+            age_hist_data=age_hist_data,
+        )
+    )
+
+    # Add variant quality histograms to info dict.
+    vcf_info_dict.update(make_hist_dict(bin_edges, adj=True))
+
+    # Add in silico prediction annotations to info_dict.
+    vcf_info_dict.update(in_silico_dict)
+
+    # Add VRS annotations to info_dict.
+    vcf_info_dict.update(vrs_fields_dict)
+
+    return vcf_info_dict
+
+
+def prepare_vcf_header_dict(
+    ht: hl.Table,
+    validated_ht: hl.Table,
+    bin_edges: Dict[str, str],
+    age_hist_data: str,
+    subset_list: List[str],
+    pops: Dict[str, str],
+    format_dict: Dict[str, Dict[str, str]] = FORMAT_DICT,
+) -> Dict[str, Dict[str, str]]:
+    """
+    Prepare VCF header dictionary.
+
+    :param ht: Input Table
+    :param validated_ht: Validated HT with unfurled info fields.
+    :param bin_edges: Dictionary of variant annotation histograms and their associated bin edges.
+    :param age_hist_data: Pipe-delimited string of age histograms, from `get_age_distributions`.
+    :param subset_list: List of sample subsets in dataset.
+    :param pops: List of sample global genetic ancestry group names for gnomAD genomes.
```

Comment:
Same comment about saying "gnomAD genomes", I think we should remove it, so we can add later for genomes.

Reply:
I'll flip to data type since the pop list is currently specific to data type